wallet/src/signer/trezor_signer/mod.rs

```diff
@@ -348,6 +381,55 @@ impl TrezorSigner {
 
         Ok((current_signatures, status))
     }
+
+    fn sign_with_standalone_private_keys(
+        &self,
+        current_signatures: AuthorizedClassicalMultisigSpend,
+        standalone_inputs: &[StandaloneInput],
+        new_status: SignatureStatus,
+        sighash: H256,
+    ) -> SignerResult<(AuthorizedClassicalMultisigSpend, SignatureStatus)> {
+        let challenge = current_signatures.challenge().clone();
+
+        standalone_inputs.iter().try_fold(
+            (current_signatures, new_status),
+            |(mut current_signatures, mut status), inp| -> SignerResult<_> {
+                if status == SignatureStatus::FullySigned {
+                    return Ok((current_signatures, status));
+                }
+
+                let key_index = inp
+                    .multisig_idx
+                    .ok_or(PartiallySignedMultisigStructureError::InvalidSignatureIndex)?;
```

Comment:
`InvalidSignatureIndex` currently has the message "The challenge contains a signature for a public key that does not exist", which is not applicable to the current situation.
I suggest either making the current message as generic as the error name `InvalidSignatureIndex` itself, or (better) renaming `InvalidSignatureIndex` to something more specific (so that it better reflects the existing message) and adding a new error variant specific for the current case.